### PR TITLE
migrate api server

### PR DESCRIPTION
**Pull request by g1nt0ki**

### CODE REVIEW

File: api2/routes/getChainDominance.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      totalCirculating: tokenBalance;
+      totalCirculatingUSD: tokenBalance;
+      greatestMcap: {
+        gecko_id: string;
+        symbol: string;
+        mcap: number;
+      };
+    };
+  };
+  // quick fix; need to update later
```
Comment (Define101) [inline api2/routes/getChainDominance.ts:22]: Should also included this ones? https://github.com/DefiLlama/peggedassets-server/blob/master/src/utils/normalizeChain.ts#L1

Comment (Define101) [reply api2/routes/getChainDominance.ts:22]: They are not that many, want me to add an if for each case?

File: api2/cache.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+} = {}
+
+const MINUTES = 60 * 1000
+const HOUR = 60 * MINUTES
+
+export async function initCache() {
+  console.time('Cache initialized')
+  const _cache = await readFromPGCache('cron-cache') ?? {}
+  Object.keys(_cache).forEach(key => cache[key] = _cache[key])
+  cache.rates = await await getLastRecord(historicalRates);
```
Comment (0xngmi) [inline api2/cache.ts:33]: await await?

File: api2/cron-task/storeCharts.ts [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+
+type TokenBalance = {
+  [token: string]: number | undefined;
+};
+
+
+export default async function handler() {
+  const timeKey = "chart-update-init";
+  console.time(timeKey);
+  const historicalRates = await (
```
Comment (0xngmi) [inline api2/cron-task/storeCharts.ts:31]: unnecessary await here

File: api2/cron-task/storePeggedPrices.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  // store bridge info (name, links) in s3
+  await store("bridgeInfo.json", JSON.stringify(bridgeInfo))
+
+  const hourTimestamp = getCurrentUnixTimestamp()
+  const dayTimestamp = getTimestampAtStartOfDay(hourTimestamp)
+
+  // store hourly prices in db
+  await dynamodb.put({ PK: hourlyPeggedPrices, SK: hourTimestamp, prices: prices, })
+  await dynamodb.put({ PK: dailyPeggedPrices, SK: dayTimestamp, prices: prices, })
```
Comment (0xngmi) [inline api2/cron-task/storePeggedPrices.ts:27]: isnt this overwriting daily prices all the time?

**Review by 0xngmi — APPROVED** [review]
lgtm overall, i didn't give it a very in-depth review but overall good

while at it can you make sure to delete history for UST after depeg and avoid storing more data for it? (maybe just change adapter to return 0?)

issue with it is that after it became a memecoin it has been messing up our list a lot since it has some weird volatility, however we can't simply delist it since historically it contributed a lot to global stablecoin marketcap and removing historical data for it would lead to global historical chart being wrong

File: api2/routes/getChainDominance.ts (hunk 1 above)
Comment (g1nt0ki) [reply api2/routes/getChainDominance.ts:22]: good idea but lets test/merge this first, PR is already quite big

Can you add it as a task to do after this?

File: api2/cache.ts (hunk 2 above)
Comment (g1nt0ki) [reply api2/cache.ts:33]: javascript is very forgiving

File: api2/cron-task/storePeggedPrices.ts (hunk 4 above)
Comment (g1nt0ki) [reply api2/cron-task/storePeggedPrices.ts:27]: yeah, it is intentional. dont think it is an issue.

Comment (0xngmi) [reply api2/cron-task/storePeggedPrices.ts:27]: its okay but afaik before we werent overwriting so previous daily data will be at the start of the day while new data will be at the end of the day, so there will be one day with 48h difference in data points

**Pull request merged**